File: 4기 코딩테스트_220615/4기 - 1st 코딩테스트/TruncateLetterArray/TruncateLetterArray_test.js
import assert from 'assert'
import solution from './TruncateLetterArray.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = ['We can', 'give advice', 'but we cannot give conduct']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(['We', 'advice', 'but', 'can', 'cannot', 'conduct', 'give', 'we'], result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = ['Hello', 'world', 'Nice World']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(['Hello', 'Nice', 'World', 'world'], result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = ['Hi', 'My Name', 'Is dave']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(['Hi', 'Is', 'My', 'Name', 'dave'], result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = ['Better late', 'than never']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(['Better', 'late', 'never', 'than'], result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = ['Slow', 'and', 'steady', 'win the race']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(['Slow', 'and', 'race', 'steady', 'the', 'win'], result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = ['Rome is', 'not built', 'in a day']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(['Rome', 'a', 'built', 'day', 'in', 'is', 'not'], result)
    done();
  });
})

File: 4기 코딩테스트_220615/4기 - 1st 코딩테스트/TruncateLetterArray/TruncateLetterArray_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.hard.TruncateLetterArray.TruncateLetterArray import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = ['We can', 'give advice', 'but we cannot give conduct']

        # when
        result = solution(arr)

        # then
        self.assertEqual(['We', 'advice', 'but', 'can', 'cannot', 'conduct', 'give', 'we'], result)

    def test_2(self):
        # given
        arr = ['Hello', 'world', 'Nice World']

        # when
        result = solution(arr)

        # then
        self.assertEqual(['Hello', 'Nice', 'World', 'world'], result)

    def test_3(self):
        # given
        arr = ['Hi', 'My Name', 'Is dave']

        # when
        result = solution(arr)

        # then
        self.assertEqual(['Hi', 'Is', 'My', 'Name', 'dave'], result)

    def test_4(self):
        # given
        arr = ['Better late', 'than never']

        # when
        result = solution(arr)

        # then
        self.assertEqual(['Better', 'late', 'never', 'than'], result)

    def test_5(self):
        # given
        arr = ['Slow', 'and', 'steady', 'win the race']

        # when
        result = solution(arr)

        # then
        self.assertEqual(['Slow', 'and', 'race', 'steady', 'the', 'win'], result)

    def test_6(self):
        # given
        arr = ['Rome is', 'not built', 'in a day']

        # when
        result = solution(arr)

        # then
        self.assertEqual(['Rome', 'a', 'built', 'day', 'in', 'is', 'not'], result)


if __name__ == '__main__':
    unittest.main()
